homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -1,12 +1,273 @@
 """Config flow for RFXCOM RFXtrx integration."""
 import logging
 
+import voluptuous as vol
+
 from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_COMMAND_OFF,
+    CONF_COMMAND_ON,
+    CONF_DEVICE,
+    CONF_DEVICE_ID,
+    CONF_DEVICES,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import (
+    async_entries_for_config_entry,
+    async_get_registry,
+)
 
-from . import DOMAIN
+from . import DOMAIN, get_device_id, get_rfx_object
+from .binary_sensor import supported as binary_supported
+from .const import (
+    CONF_AUTOMATIC_ADD,
+    CONF_DATA_BITS,
+    CONF_DEBUG,
+    CONF_FIRE_EVENT,
+    CONF_OFF_DELAY,
+    CONF_SIGNAL_REPETITIONS,
+    DEVICE_PACKET_TYPE_LIGHTING4,
+)
+from .cover import supported as cover_supported
+from .light import supported as light_supported
+from .switch import supported as switch_supported
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_OFF_DELAY_ENABLED = CONF_OFF_DELAY + "_enabled"
+CONF_REMOVE_DEVICE = "remove_device"
+
+
+class OptionsFlow(config_entries.OptionsFlow):
+    """Handle Rfxtrx options."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize rfxtrx options flow."""
+        self._config_entry = config_entry
+        self._global_options = None
+        self._selected_device = None
+        self._selected_device_event_code = None
+        self._rfxobj = None
+        self._device_entries = None
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        return await self.async_step_prompt_options()
+
+    async def async_step_prompt_options(self, user_input=None):
+        """Prompt for options."""
+        errors = {}
+
+        if user_input is not None:
+            self._global_options = {
+                CONF_DEBUG: user_input[CONF_DEBUG],
+                CONF_AUTOMATIC_ADD: user_input[CONF_AUTOMATIC_ADD],
+            }
+            if CONF_DEVICE in user_input:
+                event_code = self._get_event_code(user_input[CONF_DEVICE])
+                if not event_code:
+                    errors = {"base": "unknown_event_code"}
+                else:
+                    self._selected_device_event_code = event_code
+                    self._selected_device = self._config_entry.data[CONF_DEVICES][
+                        event_code
+                    ]
+                    return await self.async_step_set_device_options()
+            if CONF_REMOVE_DEVICE in user_input:
+                event_code = self._get_event_code(user_input[CONF_REMOVE_DEVICE])
+                if not event_code:
+                    errors = {"base": "unknown_event_code"}
+                else:
+                    device_registry = await async_get_registry(self.hass)
+                    device_registry.async_remove_device(user_input[CONF_REMOVE_DEVICE])
```

Comment:
This needs some work. This removes the device and entities, but the platform doesn't know it's been removed so it will try to unload entities when async_entry_unload is called.

Reply:
Maybe move the clean up to the dispatch target instead?

Reply:
The dispatch target(s) are the entities and directly calls `self.async_remove` on the entities.

To have it in a dispatch target I'd have to add another listener for that during setup of the entry (and remove it during unload), which is why I did it here. Do you see that as a better solution?